inc/item_devices.class.php

```diff
@@ -926,6 +926,19 @@ function getTableGroup(CommonDBTM $item, HTMLTableMain $table, array $options,
                         $content = Dropdown::getDropdownName($dropdownType::getTable(), $link[$field]);
                         break;
 
+                     case 'progressbar':
+                        $percent = 0;
+                        if ($peer->fields[$attributs['full']] > 0) {
+                           $percent = round(100 * $this->fields[$field] / $peer->fields[$attributs['full']]);
+                        }
```

Comment:
Not sure about that, but it may help to display a progressbar without requireing a field in peer item, if field value is already in percentages.
```suggestion
                        $max = array_key_exists('max', $attributs) ? $peer->fields[$attributs['max']] : 100;
                        if ($max > 0) {
                           $percent = round(100 * $this->fields[$field] / $max);
                        }
```

Reply:
Agent sends an absolute value, and GLPI expects usually an absolute value. 
Peer is already loaded earlier in the method, then why not ise it ? 

$atributs contains metadata, close to a search option. I think that the max value should not (and probably cannot) be placed here.

Reply:
I was talking about being able to use a progressbar for something else in any device, something generci, that could already been set in percentage in DB.

Reply:
it should be already usable with the current code. Just need to set datatype to 'progressbar' and use a 'max' key in item_deviceFOO.class.php . the max key will specify the column to pick in deviceFOO